lib/identity_cache/query_api.rb

```diff
@@ -508,21 +513,7 @@ def set_inverse_of_cached_association(association_options, association_target)
     end
 
     def expire_primary_index # :nodoc:
-      return unless self.class.primary_cache_index_enabled
-
-      IdentityCache.logger.debug do
-        extra_keys =
-          if respond_to?(:updated_at)
-            old_updated_at = old_values_for_fields([:updated_at]).first
-            "expiring_last_updated_at=#{old_updated_at}"
-          else
-            ""
-          end
-
-        "[IdentityCache] expiring=#{self.class.name} expiring_id=#{id} #{extra_keys}"
-      end
-
-      IdentityCache.cache.delete(primary_cache_index_key)
+      self.class.send(:expire_primary_index, id)
```

Comment:
Maybe we should make `expire_primary_index` and `expire_cache` public to make it clearer that we depend on them in our application, so they aren't really just internal library methods.

Reply:
I opted to keep them private so that other objects can't use them. Maybe protected could make sense here if the intent is to make it clear that they are safe to use outside of the library.

Reply:
We would still need to use `self.class.send(:expire_primary_index, id)` even if the method were `protected`.  I'm not sure if makes sense to keep methods like `expire_cache` private given that active record already exposes lots of low level public methods on active record objects (e.g. `delete`, `update_column`).

Reply:
I've made `expire_primary_index` public, I can make `expire_cache` public in this PR too if you want.

Reply:
> I can make expire_cache public in this PR too if you want

Yes, please

Reply:
Should it make `expire_attribute_indexes` and the instance method `expire_primary_index` public too? I just made `expire_cache` in my additional commit but I can move all of them.

Reply:
If they have an instance, then it seems preferable for them to use `expire_cache`